server/src/logic/utils/model.utils.ts

```diff
@@ -0,0 +1,19 @@
+import { ApiError, ERRORS } from "./../../models/index";
+
+export class ModelUtils {
```

Comment:
The model.utils takes a existing object (like a recipe already stored in the db) and a new value  (obj which contains properties which should be updated in the existing value). Then it checks if the provided properties are available and updatable in the existing object (the existing obj has a map with it's updatable fields). If the fields are available and updatable, these fields will be overwritten with the new values. The updated object will be returned. Therefor I cannot use Object.assign since this would also add new properties to the existing object if they do not exist already.